src/api/createDeviceMessageHandlingSystem/src/v1/steps.py

```diff
@@ -36,7 +111,7 @@ def validate_spine_mhs_questionnaire_response(data, cache) -> QuestionnaireRespo
     questionnaire_responses = payload.questionnaire_responses
 
     # Ensure there's a questionnaire named 'spine_mhs' in the responses
-    if QuestionnaireInstance.SPINE_MHS not in questionnaire_responses:
+    if "spine_mhs" not in questionnaire_responses:
```

Comment:
2 questions...

* why switch from using the constant? 
* i think maybe we got the logic slightly wrong the first time we wrote this - should it not be:

```suggestion
    if questionnaire_responses.keys() != {QuestionnaireInstance.SPINE_MHS}:
```

since we wouldn't want to allow any other questionnaire to be sent as well

Reply:
Added this logic into the incoming params to be flagged at the initial stage